cv/lane_detection/src/lane_detection_model_inference.py

```diff
@@ -1,136 +1,143 @@
+        shape = struct.pack(">II", h, w)
 
-        retval, buffer = cv2.imencode('.png', img)
+        retval, buffer = cv2.imencode(".png", img)
         img_bytes = np.array(buffer).tostring()
 
         encoded = shape + img_bytes
 
         # Store encoded data in Redis
-        self.redis.set(name,encoded)
+        self.redis.set(name, encoded)
 
         return
 
+
 def find_edge_channel(img):
-    edges_mask = np.zeros((img.shape[0],img.shape[1]),dtype=np.uint8)
+    edges_mask = np.zeros((img.shape[0], img.shape[1]), dtype=np.uint8)
     width = img.shape[1]
     height = img.shape[0]
 
-    gray_im = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
+    gray_im = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
     # gray_im = cv2.GaussianBlur(gray_im,(3,3),0)
     # Separate into quadrants
-    med1 = np.median(gray_im[:height//2,:width//2])
-    med2 = np.median(gray_im[:height//2,width//2:])
-    med3 = np.median(gray_im[height//2:,width//2:])
-    med4 = np.median(gray_im[height//2:,:width//2])
+    med1 = np.median(gray_im[: height // 2, : width // 2])
+    med2 = np.median(gray_im[: height // 2, width // 2 :])
+    med3 = np.median(gray_im[height // 2 :, width // 2 :])
+    med4 = np.median(gray_im[height // 2 :, : width // 2])
 
-    l1 = int(max(0,(1-0.205)*med1))
-    u1 = int(min(255,(1+0.205)*med1))
-    e1 = cv2.Canny(gray_im[:height//2,:width//2],l1,u1)
+    l1 = int(max(0, (1 - 0.205) * med1))
+    u1 = int(min(255, (1 + 0.205) * med1))
+    e1 = cv2.Canny(gray_im[: height // 2, : width // 2], l1, u1)
 
-    l2 = int(max(0,(1-0.205)*med2))
-    u2 = int(min(255,(1+0.205)*med2))
-    e2 = cv2.Canny(gray_im[:height//2,width//2:],l2,u2)
+    l2 = int(max(0, (1 - 0.205) * med2))
+    u2 = int(min(255, (1 + 0.205) * med2))
+    e2 = cv2.Canny(gray_im[: height // 2, width // 2 :], l2, u2)
 
-    l3 = int(max(0,(1-0.205)*med3))
-    u3 = int(min(255,(1+0.205)*med3))
-    e3 = cv2.Canny(gray_im[height//2:,width//2:],l3,u3)
+    l3 = int(max(0, (1 - 0.205) * med3))
+    u3 = int(min(255, (1 + 0.205) * med3))
+    e3 = cv2.Canny(gray_im[height // 2 :, width // 2 :], l3, u3)
 
-    l4 = int(max(0,(1-0.205)*med4))
-    u4 = int(min(255,(1+0.205)*med4))
-    e4 = cv2.Canny(gray_im[height//2:,:width//2],l4,u4)
+    l4 = int(max(0, (1 - 0.205) * med4))
+    u4 = int(min(255, (1 + 0.205) * med4))
+    e4 = cv2.Canny(gray_im[height // 2 :, : width // 2], l4, u4)
 
     # Stitch the edges together
-    edges_mask[:height//2,:width//2] = e1
-    edges_mask[:height//2,width//2:] = e2
-    edges_mask[height//2:,width//2:] = e3
-    edges_mask[height//2:,:width//2] = e4
+    edges_mask[: height // 2, : width // 2] = e1
+    edges_mask[: height // 2, width // 2 :] = e2
+    edges_mask[height // 2 :, width // 2 :] = e3
+    edges_mask[height // 2 :, : width // 2] = e4
 
     edges_mask_inv = cv2.bitwise_not(edges_mask)
 
     return edges_mask, edges_mask_inv
 
 
-def get_input(frame):
-    frame = cv2.resize(frame,(1280,720),interpolation=cv2.INTER_AREA)
```

Comment:
I deleted this because it makes the model do work on an unnecessarily large image